powerauth-push-server/src/main/java/io/getlime/push/service/PushMessageSenderService.java

```diff
@@ -278,6 +288,22 @@ private void updateStatusAndPersist(PushMessageEntity pushMessageObject, PushMes
         }
     }
 
+    private ApnsEnvironment resolveApnsEnvironment(final String environmentDevice) {
+        if (environmentDevice != null) {
+            final ApnsEnvironment envDevice = ApnsEnvironment.fromString(environmentDevice);
+            if (!configuration.isApnsUseDevelopment() && envDevice == ApnsEnvironment.DEVELOPMENT) {
+                logger.warn("Invalid configuration: server is configured with APNs in production mode, however push device is registered in development mode.");
+                return null;
+            }
+            return envDevice;
+        }
+        // Fallback in case device was registered without environment
+        if (configuration.isApnsUseDevelopment()) {
```

Comment:
and we still have the config on the application level

Reply:
Added.